lib/Bitcode/Writer/BitcodeWriter.cpp

```diff
@@ -2019,10 +2019,27 @@ static void WriteValueSymbolTable(const ValueSymbolTable &VST,
   // FIXME: We know if the type names can use 7-bit ascii.
   SmallVector<unsigned, 64> NameVals;
 
+// HLSL Change - Begin
+  // Read the named values from a sorted list instead of the original list
+  // to ensure the binary is the same no matter what values ever existed.
+  SmallVector<const ValueName *, 16> SortedTable;
+
   for (ValueSymbolTable::const_iterator SI = VST.begin(), SE = VST.end();
        SI != SE; ++SI) {
+    SortedTable.push_back(&(*SI));
+  }
+  std::sort(SortedTable.begin(), SortedTable.end(), [](const ValueName *A, const ValueName *B) {
+    return (*A).first() < (*B).first();
+  });
```

Comment:
This set off some red flags from my memory of an esoteric bug where the differing platform implementations of std::sort would result in inconsistent results if a tie was encountered. That ordering also determined a signature.

 As far as I can tell, this map rejects duplicate keys. So that shouldn't be a problem here. I've not closely examined any other part of the change, but in case someone else was worried, I don't think it's an issue.

Reply:
#1470 if you're curious.

Reply:
Thanks for the info! I added a comment to address this.